src/worker/prettier-instance-worker.js

```diff
@@ -0,0 +1,108 @@
+const { parentPort } = require("worker_threads");
+
+const path2ModuleCache = new Map();
+
+parentPort.on("message", ({ type, payload }) => {
+  switch (type) {
+    case "import": {
+      const { modulePath } = payload;
+      let prettierInstance = path2ModuleCache.get(modulePath);
+      if (!prettierInstance) {
+        try {
+          prettierInstance = require(modulePath);
+          // If the instance is missing `format`, it's probably
+          // not an instance of Prettier
+          if (!prettierInstance.format) {
+            parentPort.postMessage({
+              type,
+              payload: { version: null },
+            });
+            break;
+          }
+          path2ModuleCache.set(modulePath, prettierInstance);
+        } catch {
+          parentPort.postMessage({
+            type,
+            payload: { version: null },
+          });
+          break;
+        }
+      }
+      parentPort.postMessage({
+        type,
+        payload: { version: prettierInstance.version },
+      });
+      break;
+    }
+    case "callMethod": {
+      const { modulePath, methodName, methodArgs, id } = payload;
+      const prettierInstance = path2ModuleCache.get(modulePath);
+      if (!prettierInstance) {
+        parentPort.postMessage({
+          type,
+          payload: {
+            result: new Error(
+              "Cannot load prettier instance from cache. Please call import() before call callMethod."
```

Comment:
We have `modulePath` in `PrettierWorkerInstance` we can pass it every time or set it once, and `import` it first here.